app/models/push_notification_target.rb

```diff
@@ -0,0 +1,113 @@
+class PushNotificationTarget < ApplicationRecord
+  resourcify
+
+  FOREM_BUNDLE = "com.forem.app".freeze
+  SUPPORTED_PLATFORMS = [Device::ANDROID, Device::IOS].freeze
+  FOREM_APP_PLATFORMS = [Device::IOS].freeze
+
+  validates :app_bundle, presence: true
+  validates :platform, presence: true
+
+  # Clear Redis-backed model to ensure it will be recreated with updated values
+  after_update :clear_rpush_app
+
+  def forem_app?
+    app_bundle == FOREM_BUNDLE
+  end
+
+  def active?
+    # TRUE if it's marked as `active` in the DB && it has credentials available
+    active && auth_credentials?
+  end
+
+  def auth_credentials?
+    auth_credentials.present?
+  end
+
+  def auth_credentials
+    if app_bundle == FOREM_BUNDLE && platform == Device::IOS
+      ApplicationConfig["RPUSH_IOS_PEM"]
+    else
+      auth_key
+    end
+  end
```

Comment:
The Forem apps will get their credentials from an ENV variable, whereas custom PN targets will get their credentials from the `auth_key` stored in the DB (configured by the Forem creator).

Reply:
+1 on transforming this as a comment :D